drivers/modem/hl7800.c

```diff
@@ -1905,11 +1905,14 @@ static void dns_work_cb(struct k_work *work)
 				 &temp_addr);
 	if (!valid_address && IS_ENABLED(CONFIG_NET_IPV4)) {
 		/* IPv6 DNS string is not valid, replace it with IPv4 address and recheck */
-		strncpy(iface_ctx.dns_v6_string, iface_ctx.dns_v4_string,
```

Comment:
The `n` in `strncpy` is the size of the destination, I think the fix should be
```
strncpy(iface_ctx.dns_v6_string, iface_ctx.dns_v4_string, sizeof(iface_ctx.dns_v6_string) - 1);
```
given that `iface_ctx.dns_v6_string` is initialized to `0` it will always be terminated